### ISSUE TITLE
Swin in this repo + dynamic resolution

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/vision#6227]
### 📚 The doc issue

Does Swin impl in this repo support arbitrary dynamic execution-time-defined input resolution (same as other backbones)?

Initially Swin was trained to support only one resolution, but then hacks can be done to support arbitrary resolution. Two repos with such hacks:
* https://github.com/SwinTransformer/Swin-Transformer-Object-Detection/blob/master/mmdet/models/backbones/swin_transformer.py
* https://github.com/megvii-research/SOLQ/blob/main/models/swin_transformer.py

Related issues: https://github.com/microsoft/SimMIM/issues/13 https://github.com/microsoft/esvit/issues/17

cc @YosuaMichael

**Comment by YosuaMichael** [comment]
Hi @vadimkantorov , from my understanding the answer is yes. 

The swin_transformer implementation in TorchVision dont restrict the input_resolution. However take note that the accuracy may changes depending on your input resolution (or transform).

For example, the original accuracy of swin_transformer using `resize_size=232` and `crop_size=224` achieve `Acc@1 81.472 Acc@5 95.780` in imagenet1k. I have tried to run the validation with `resize_size=128` and `crop_size=112` on the same dataset and I got: `Acc@1 69.248 Acc@5 88.824`.

**Comment by vadimkantorov** [comment]
well, `resize_size` makes think that it's resizing the input instead and always gives a fixed output resolution

from what I understand, the two impl I mentioned above allow for scaling output resolution depending on input resolution

**Comment by vadimkantorov** [comment]
at the very least, the actual behavior should be clearly explained in the docs...

**Comment by YosuaMichael** [comment]
> well, resize_size makes think that it's resizing the input instead and always gives a fixed output resolution
> 
> from what I understand, the two impl I mentioned above allow for scaling output resolution depending on input resolution
> 

Hi @vadimkantorov , just to clarify, the `resize_size` that I mentioned is the `transform` before we do the classification task. It is not part of the `swin_transformer` parameters (sorry for the confusion!)).

For `output resolution` can you explain more what do you mean by it?
From what I understand now the swin transformer model on TorchVision is a meant for classification model, so it will definitely output fixed size output according to the num_classes parameter.

For the two links that you provide, I think their implementation are focus as a backbone and the model return the output of all the layers.

**Comment by vadimkantorov** [comment]
I think for all models in torchvision, being able to serve as backbone is top-1 or top-2 usage anyway. People usually figure out how to remove the classifier, and just use the features of some layer. So this is important to clarify in docs and important scenario in general. Those two repos above achieve this by hacking the architecture a bit

**Comment by vadimkantorov** [comment]
Probably this repo/detectron2 also has swin adapted as a backbone: https://github.com/facebookresearch/detectron2/tree/main/projects/ViTDet
